src/datasources/accounts/accounts.datasource.ts

```diff
@@ -64,4 +66,50 @@ export class AccountsDatasource implements IAccountsDatasource {
   async getDataTypes(): Promise<AccountDataType[]> {
     return this.sql<[AccountDataType]>`SELECT * FROM account_data_types`;
   }
+
+  /**
+   * Adds or updates the existing account data settings for a given address/account.
+   * Requirements:
+   * - The account must exist.
+   * - The data type must exist.
+   * - The data type must be active.
+   *
+   * @param address - account address.
+   * @param upsertAccountDataSettings {@link UpsertAccountDataSettingsDto} object.
+   * @returns {Array<AccountDataSetting>} inserted account data settings.
+   */
+  async upsertAccountDataSettings(
+    address: `0x${string}`,
+    upsertAccountDataSettings: UpsertAccountDataSettingsDto,
+  ): Promise<AccountDataSetting[]> {
+    const account = await this.getAccount(address);
+    const dataTypes = await this.getDataTypes();
+    const { accountDataSettings } = upsertAccountDataSettings;
+
+    const result = [];
+    for (const ads of accountDataSettings) {
+      const dataType = dataTypes.find((dt) => dt.name === ads.dataTypeName);
+      if (!dataType) {
+        throw new UnprocessableEntityException('Invalid data type.');
+      }
+      if (!dataType.is_active) {
+        throw new UnprocessableEntityException(
+          `Data type ${dataType.name} is inactive.`,
+        );
+      }
+      const [inserted] = await this.sql<[AccountDataSetting]>`
+        INSERT INTO account_data_settings (account_id, account_data_type_id, enabled)
+        VALUES (${account.id}, ${dataType.id}, ${ads.enabled})
+        ON CONFLICT (account_id, account_data_type_id) DO UPDATE SET enabled = EXCLUDED.enabled
+        RETURNING *
+      `.catch((e) => {
+        throw new UnprocessableEntityException(
+          `Error updating data settings: ${asError(e).message}`,
+        );
+      });
```

Comment:
Would there be any benefit in doing these in one transaction?

Reply:
The transaction was included in [76e14ea](https://github.com/safe-global/safe-client-gateway/pull/1715/commits/76e14ea2747ff81e473d2b4e1916e85619a06c04)